components/ProfileDropdownMenu.tsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react'
+import Dropdown from 'react-bootstrap/Dropdown'
+import UserInfoImage from './UserInfoImage'
+import { ChevronDownIcon } from '@primer/octicons-react'
+import LogoutContainer from './LogoutContainer'
+import { Button } from './theme/Button'
+import { UserInfo } from '../@types/user'
+import { useUserInfoQuery } from '../graphql/index'
+import { useRouter } from 'next/router'
+import _ from 'lodash'
+import styles from '../scss/profileDropDown.module.scss'
+
+type ProfileDropDownMenuProps = {
+  username: string
+  isAdmin: boolean
+}
+
+type CustomToggleProps = {
+  children?: React.ReactNode
+  className?: string | undefined
+  onClick?: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {}
+}
+
+const ProfileDropdownMenu: React.FC<ProfileDropDownMenuProps> = ({
+  username,
+  isAdmin
+}) => {
+  const { data } = useUserInfoQuery({
+    variables: { username },
+    skip: !username
+  })
+  const router = useRouter()
+  const location = router.asPath
+
+  const fullname = _.get(data, 'userInfo.user.name', '')
```

Comment:
Why do you use lodash `get` here, but elsewhere you use optional chaining?

Reply:
Yeah let's be consistent.

Reply:
Use lodash `get` when you want to return a value other than `undefined` when the key is not found. Optional chaining when the `undefined` value is fine.

Reply:
@flacial If we want to display username as empty string instead of undefined, then just add `||`

```
data?.userInfo?.user?.name || ''
```


IMO I prefer to use lodash `get`, but I believe some conversation was had awhile ago that this codebase should use optional chaining instead of `get`. We really shouldn't be mixing the two, its better to be consistent